pkg/order/etcdraft/util.go

```diff
@@ -109,4 +110,44 @@ func msgToConsensusPbMsg(data []byte, tyr raftproto.RaftMessage_Type, replicaID
 		Data: cmData,
 	}
 	return msg
+}
+
+func (n *Node) getSnapshot() ([]byte, error) {
+	cm := pb.ChainMeta{
+		Height: n.lastExec,
+	}
+	return cm.Marshal()
+}
+
+func (n *Node) recoverFromSnapshot()  {
+	snapshot, err := n.raftStorage.snap.Load()
+	if err != nil {
+		n.logger.Error(err)
+		return
+	}
+	targetChainMeta := &pb.ChainMeta{}
+	err = targetChainMeta.Unmarshal(snapshot.Data)
+	if err != nil {
+		n.logger.Error(err)
+		return
+	}
+
+	chainMeta := n.getChainMetaFunc()
+	n.logger.WithFields(logrus.Fields{
+		"target":       targetChainMeta.Height,
+		"current":      chainMeta.Height,
+		"current_hash": chainMeta.BlockHash.String(),
+	}).Info("State Update")
+
+	blockCh := make(chan *pb.Block, 1024)
+	go n.syncer.SyncCFTBlocks(chainMeta.Height+1, targetChainMeta.Height, blockCh)
+	for block := range blockCh {
+		if block == nil {
+			break
+		}
+		n.commitC <- block
+		n.lastExec = block.Height()
```

Comment:
please check if  n.lastExec is equal to the target synchronization height finally.